timed/reporting/management/commands/notify_supervisors.py

```diff
@@ -0,0 +1,126 @@
+from datetime import date, timedelta
+
+from django.conf import settings
+from django.contrib.auth import get_user_model
+from django.core.mail import send_mass_mail
+from django.core.management.base import BaseCommand
+from django.template.loader import render_to_string
+
+from timed.employment.serializers import UserSerializer
+
+
+class Command(BaseCommand):
+    """
+    Notify supervisors command.
+
+    This command provides several methods usually used in a cron job
+    to notify supervisors on tracking status of their supervisees.
+    """
+
+    help = 'Notify supervisors on tracking status of user'
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.serializer = UserSerializer()
+
+    def add_arguments(self, parser):
+        parser.add_argument('notification', choices=['shorttime'])
+        parser.add_argument('options', nargs='?')
+
+    def handle(self, *args, **options):
+        """Call given notification method and pass on options."""
+        method = getattr(self, options['notification'])
+        method(*options['options'])
+
+    def _decimal_hours(self, duration):
+        return duration.total_seconds() / 3600
+
+    def _get_supervisees_with_shorttime(self, start, end, ratio):
+        """
+        Get supervisees which reported less hours than they have should.
+
+        :return: dict mapping all supervisees with shorttime with tuple of
+                 reported, expected, balance and actual ratio.
+        """
+        supervisees_shorttime = {}
+        supervisees = get_user_model().objects.all_supervisees()
+        for supervisee in supervisees:
+            worktime = self.serializer.get_worktime(supervisee, start, end)
+            reported, expected, balance = worktime
```

Comment:
this doesn't fit into 80 characters and I would have to add a new line at the dot... not really readable hence this solution.